python/semantic_kernel/agents/agent.py

```diff
@@ -31,7 +31,7 @@ class Agent(ABC, KernelBaseModel):
 
     id: str = Field(default_factory=lambda: str(uuid.uuid4()))
     description: str | None = None
-    name: str | None = None
+    name: str = Field(default_factory=lambda: f"agent_{generate_random_ascii_name()}")
     instructions: str | None = None
     kernel: Kernel = Field(default_factory=Kernel)
     channel_type: ClassVar[type[AgentChannel] | None] = None
```

Comment:
Unclear why channel_type needs to be defined outside of `get_channel_keys` and `create_channel`.

Reply:
channel_type has not changed since the last iteration of code.